### PR TITLE
Support HTTP_PLATFORM_PORT.

### PR DESCRIPTION
**Pull request by Tratcher**
#358 @davidfowl @muratg @halter73

### CODE REVIEW

**Comment by halter73** [comment]
:shipit:

**Comment by halter73** [comment]
The ServerTests will rely on this to pass, correct?

File: src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs [hunk 1]
```diff
@@ -188,6 +191,16 @@ private RequestDelegate BuildApplication()
             var builder = builderFactory.CreateBuilder(_serverInstance);
             builder.ApplicationServices = _applicationServices;
 
+            var port = _config[ServerPort];
+            if (!string.IsNullOrEmpty(port))
+            {
+                var addresses = builder.ServerFeatures.Get<IServerAddressesFeature>()?.Addresses;
+                if (addresses != null && !addresses.IsReadOnly)
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs]: When is this ever read only.

**Comment by davidfowl** [comment]
Add a test

File: src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs (hunk 1 above)
Comment (Tratcher) [inline src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs]: If the server doesn't support modifying it at runtime. e.g. IIS/Helios.

**Comment by Tratcher** [comment]
@halter73 Yes, the new HttpPlatformHandler tests will require it.

**Comment by Tratcher** [comment]
Added test.

**Comment by davidfowl** [comment]
:shipit:

**Pull request merged**